### PR TITLE
Print Progress when running draws()

### LINKED ISSUE #121: Add some form of notification messaging to draws
Might be nice to add some form of progress notification to draws to inform the user how far through the process it is (especially for mmrm based ones where there is a lot to process.

### PR DESCRIPTION
**Pull request by gowerc**
closes #121

### CODE REVIEW

**Comment by gowerc** [comment]
@nociale , I think this is ready for review now.  I've updated all the unit tests to use the "quiet" version, I've updated method_bayes to remove verbose and instead rely on the new `quiet` option on `draws()`.

File: R/draws.R [hunk 1]
```diff
@@ -27,6 +27,8 @@
 #' It specifies the multiple imputation methodology to be used. See details.
 #' @param ncores A single numeric specifying the number of cores to use in creating the draws object. 
 #' Note that this parameter is ignored for [method_bayes()] (Default = 1).
+#' @param quiet Logical, If true will suppress printing of progress information that is printed to
```
Comment (nociale) [inline R/draws.R:30]:
```suggestion
#' @param quiet Logical, If `TRUE` will suppress printing of progress information that is printed to
```

File: R/draws.R [hunk 2]
```diff
@@ -27,6 +27,8 @@
...
+#' the console
```
Comment (nociale) [inline R/draws.R:31]:
```suggestion
#' the console.
```

File: R/draws.R [hunk 3]
```diff
@@ -263,6 +269,8 @@ draws.bmlmi <- function(data, data_ice = NULL, vars, method, ncores = 1) {
 #' what subjects should be used to derive the imputed dataset.
 #' @param failure_limit Number of failed samples that are allowed before throwing an error
 #' @param ncores Number of processes to parallise the job over
+#' @param quiet Logical, If true will suppress printing of progress information that is printed to
```
Comment (nociale) [inline R/draws.R:272]:
```suggestion
#' @param quiet Logical, If `TRUE` will suppress printing of progress information that is printed to
```

File: R/draws.R [hunk 4]
```diff
@@ -263,6 +269,8 @@ draws.bmlmi <- function(data, data_ice = NULL, vars, method, ncores = 1) {
...
+#' the console
```
Comment (nociale) [inline R/draws.R:273]:
```suggestion
#' the console.
```

File: R/draws.R [hunk 5]
```diff
@@ -657,3 +680,81 @@ validate.draws <- function(x, ...) {
...
+        step_current = 0,
+        
+        #' @field n integer, the current number of completed iterations
+        n = 0,
+        
+        #' @field n_max integer, total number of expected iterations to be completed
+        #' acts as the denominator for calculating progress percentages
+        n_max = NULL,
+        
+        #' @field quiet logical holds wether or not to print anything
```
Comment (nociale) [inline R/draws.R:718]:
```suggestion
        #' @field quiet logical holds whether or not to print anything
```

File: R/draws.R [hunk 6]
```diff
@@ -657,3 +680,81 @@ validate.draws <- function(x, ...) {
...
+        #' @param quiet logical, sets field `quiet`
+        #' @param step real, sets field `step`
+        initialize = function(n_max, quiet = FALSE, step = 0.1) {
+            self$step = step
+            self$n_max = n_max
+            self$quiet = quiet
+        },
+        
+        #' @description
+        #' Records that n more iterations have been completed
```
Comment (nociale) [inline R/draws.R:733]:
```suggestion
        #' Records that `n` more iterations have been completed
```

File: R/mcmc.R [hunk 7]
```diff
@@ -15,6 +15,7 @@
 #' @param subjid Character vector containing the subjects IDs.
 #' @param visit Character vector containing the visit variable.
 #' @param method A `method` object as generated by [method_bayes()].
+#' @param quiet Specify whether the stan sampling log should be printed to the console
```
Comment (nociale) [inline R/mcmc.R:18]:
```suggestion
#' @param quiet Specify whether the stan sampling log should be printed to the console.
```

**Review by nociale — APPROVED** [review]

**Pull request merged**